File: server/Model/Fandom.ts
import { Column, Entity, JoinTable, ManyToMany, OneToMany, PrimaryGeneratedColumn } from "typeorm";
import { Editor } from "./Editor";
import { Edit } from "./Edit";

@Entity()
export class Fandom {
    @PrimaryGeneratedColumn()
    fandom_id: number;

    @Column()
    fandom_name: string

    @OneToMany(() => Edit, edit => edit.edit_fandom)
    fandom_edits: Edit[]

    @ManyToMany(() => Editor, editor => editor.editor_fandom)
    @JoinTable()
    fandom_editors: Editor[];

    @Column()
    fandom_img: string
}

File: server/src/main/java/laurriana/time2day/model/Fandom.java
package laurriana.time2day.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Fandom {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int fandomId;

    private String fandomName;

    private String fandomImg;

    @OneToMany
    private ArrayList<Edit> fandomEdits;

    @ManyToMany
    private ArrayList<Editor> fanodmEditors;

}
